### ISSUE TITLE
Add Support to Symlinks on `baseFolders`

### ISSUE THREAD

**Issue by daronmcintosh** [alefragnani/vscode-project-manager#274]
#### Description

I have a folder "C:/Symbolic Link/Github Projects" that is symlinked to another folder that contains the actual projects. Opening "C:/Symbolic Link/Github Projects" in VSCode lists all the projects perfectly fine. But when I use this folder for `projectManager.git.baseFolders: ["C:/Symbolic Link/Github Projects"]` or `projectManager.any.baseFolders: ["C:/Symbolic Link/Github Projects"]` setting, project manager is unable to auto detect the projects. Auto detection works fine if I use the actual path the projects are in and not the symlinked folder. I assumed that the `fs` module from node is being used to detect the folders so I tested to see if `fs.readdir` would see the folders in the symlinked folder and it did. Are symlinked folders disabled by default for auto detection?

#### VSCode

Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17134
Extension Version: 10.5.1

**Comment by alefragnani** [comment]
Hi @iSolutionJA ,

I don't use symlinks very often. In fact, I **avoid** to use it because _there is always some tool that will fail with  it_ 😄  .

I use `walker` to detect the folders, and _it appears_ that it recognizes symlinked folders, but using it in a small test, it didn't work perfectly. It recognized the top level symliked folder, but not its subfolders. So, I have to make a few more tests.

Hope this helps

**Comment by iSolutionJA** [comment]
I recently started using them because they are great for syncing configurations between my cloud storage and keeping the paths the same between my laptop and PC.

I should've checked `package.json` but I didn't because I thought all the implementation details were probably hidden in the `vscode-project-manager-core` repo. I'll run some tests on my end to see how well `walker` detects the symlinked folders.

**Comment by iSolutionJA** [comment]
I looked into the code for `walker` and for symlinked folders it doesn't enter them. `walker` only enters folders it detects as directories, see [here](https://github.com/daaku/nodejs-walker/blob/3f78f80d026fd1efd2adc2f8927a0a4ccae35644/lib/walker.js#L57). I copied lines 58-75 and used it in the if statement for `stat.isSymbolicLink()` and replaced `dir` with `symlink` and it worked perfectly fine. It entered symlinked folders and emitted and error when it was a symlinked file.
